Tasks/Common/azure-blobstorage-downloader/azureBlobStorageProvider.ts

```diff
@@ -0,0 +1,139 @@
+import path = require('path');
+import fs = require('fs');
+import azureStorage = require('azure-storage');
+import models = require('item-level-downloader/Models');
+import Stream = require("stream");
+
+export class AzureBlobProvider implements models.IArtifactProvider {
+    constructor(storageAccount: string, container: string, accessKey: string, prefixFolderPath?: string) {
+        this._storageAccount = storageAccount;
+        this._accessKey = accessKey;
+        this._container = container;
+        this._prefixFolderPath = prefixFolderPath;
+        this._blobSvc = azureStorage.createBlobService(this._storageAccount, this._accessKey);
```

Comment:
How will this work when accessing storage accounts of National cloud etc

Reply:
adding host as a parameter that should be passed on to the blobservice for use.